notebook/services/contents/tests/test_manager.py

```diff
@@ -528,3 +528,34 @@ def test_check_and_sign(self):
         cm.mark_trusted_cells(nb, path)
         cm.check_and_sign(nb, path)
         assert cm.notary.check_signature(nb)
+
+
+class TestContentsManagerNoAtomic(TestCase):
+
+    def setUp(self):
+        self._temp_dir = TemporaryDirectory()
```

Comment:
If you do `self.addCleanup(temp_dir.cleanup)`, then you don't need a  `tearDown` method, and you don't need to attach the temporary directory object to `self`.